atom/browser/api/atom_api_app.cc

```diff
@@ -810,6 +812,16 @@ void App::ChildProcessLaunched(int process_type, base::ProcessHandle handle) {
       base::ProcessMetrics::CreateProcessMetrics(
           handle, content::BrowserChildProcessHost::GetPortProvider()));
 #else
+  // We do the same check here as what CreateProcessMetrics DCHECKs for,
```

Comment:
I'm not sure about this fix, but for some reason that [DCHECK](https://chromium.googlesource.com/chromium/src/+/66.0.3359.181/base/process/process_metrics_win.cc#332) I mentioned gets triggered, but only in this specific scenario. Does anyone have any objections against this? @electron/reviewers

Reply:
Can you identify the actual cause of the dcheck failure, working around it shouldn't be an option unless its an intentional failure caused by our implementation.

Reply:
@deepak1556 It looks like that `DCHECK` gets triggered when I add `app.disableHardwareAcceleration()`, which causes the `GPU` process to die (it can get started for a split second), and I think it dies before the `ProcessMetrics` object gets created for it, hence triggering the check for a valid process handle. I found out that if I use `--disable-gpu` instead of just `app.disableHardwareAcceleration()`, the `GPU` process never even starts, and this failure is not triggered.
Would it be fine to make `app.disableHardwareAcceleration()` work just like `--disable-gpu`? That would involve switching [this line ](https://github.com/electron/electron/blob/9f328abe19186af292eea3757ac908bc793a987c/atom/browser/api/atom_api_app.cc#L953) to be this instead:
```
auto* command_line = base::CommandLine::ForCurrentProcess();
  command_line->AppendSwitch(::switches::kDisableGpu);
```
I tested this, and it works, and should have no other consequences AFAIK, but I might be wrong.

Reply:
Thanks for the investigation! the reason api didn't use `--disable-gpu` was because the user scripts are executed only after the main message loop is initialized and by that time the flag wasn't picked up by the browser process, check `content::BrowserMainLoop`. Another way to solve this would be to avoid creating the process metric for gpu process in `App::ChildProcessLaunched` when HW acceleration is disabled. That way we are guaranteed to fix the DCHECK crash and also we could perform a cleaner destruction of the launched gpu process using `content::GpuProcessHost::ForceShutdown`. Lets do this in a separate PR.